source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -492,16 +493,18 @@ core::smart_refctd_ptr<IGPUSpecializedShader> COpenGLDriver::createGPUSpecialize
         if (!spvShader)
             return nullptr;
 
-        spv = spvShader->getSPVorGLSL();
-        spv->grab();
+        spvCPUShader = spvShader.get();
+        spvCPUShader->grab();
     }
     else {
-        spv = cpuUnspec->getSPVorGLSL();
-        spv->grab();
+        spvCPUShader = cpuUnspec;
+        spvCPUShader->grab();
```

Comment:
I don't want to see grab and drops in the engine anymore

Reply:
@Crisspl ?

Reply:
this grab is needed here

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/358#discussion_r328368814

Reply:
see my latest comment on the latest commit,

Reply:
but i think i get which one u mean